lib/vmpooler/generic_connection_pool.rb

```diff
@@ -40,13 +39,13 @@ def with_metrics(options = {})
           start = Time.now
           conn = checkout(options)
           timespan_ms = ((Time.now - start) * 1000).to_i
-          @metrics&.gauge(@metric_prefix + '.available', @available.length)
-          @metrics&.timing(@metric_prefix + '.waited', timespan_ms)
+          @metrics&.gauge("connection_available.#{@provider}", @available.length)
+          @metrics&.timing("connection_waited.#{@provider}", timespan_ms)
           begin
             yield conn
           ensure
             checkin
-            @metrics&.gauge(@metric_prefix + '.available', @available.length)
+            @metrics&.gauge("connection_available.#{@provider}", @available.length)
```

Comment:
I may be missing it, but where does the metric prefix get inserted now, particularly with non-Prometheus metrics.

Reply:
So - the variable ```metric_prefix``` is local to the connection pool and is an additional prefix over and above the general metric prefix that is defined for example in ```graphite.rb```

```metric_prefix``` is defined in the providers (e.g. see changes to ```lib/vmpooler/providers/vsphere.rb``` below or check the commit titled **(POOLER-160) Standardise Connection Metrics** 

So if you look at this in graphite, the variable for ci-next is ```vmpooler-cinext.vsphere-ci65_provider_connection_pool.available```

This code simplifies it to ```vmpooler-cinext. connection_available.vsphere```

Reply:
I think we still need this `metric_prefix`. The generic_connection_pool offers metric_prefix as a parameter to allow for a user to differentiate between metrics produced by different connection pools. #375 adds a second instance of the generic connection pool for redis connection pooling. The prefix allows the redis metrics to have 'redis_connection_pool' in front of them, to make clear when the metrics are related to redis connections, as opposed to vsphere connections.

Reply:
Can we do this using a label as all metrics for Prometheus need to be pre-registered ?
In this case I'm using the provider name as a label value.

Lets discuss further.